File: Using Workflow with ASP.NET MVC 4/[C#]-Using Workflow with ASP.NET MVC 4/C#/HelloWorkflowAndMVC/Controllers/AccountController.cs
namespace HelloWorkflowAndMVC.Controllers
{
    using System.Web.Mvc;
    using System.Web.Routing;
    using System.Web.Security;

    using HelloWorkflowAndMVC.Models;

    public class AccountController : Controller
    {
        #region Public Properties

        public IFormsAuthenticationService FormsService { get; set; }

        public IMembershipService MembershipService { get; set; }

        #endregion

        #region Public Methods and Operators

        [Authorize]
        public ActionResult ChangePassword()
        {
            this.ViewBag.PasswordLength = this.MembershipService.MinPasswordLength;
            return this.View();
        }

        [Authorize]
        [HttpPost]
        public ActionResult ChangePassword(ChangePasswordModel model)
        {
            if (this.ModelState.IsValid)
            {
                if (this.MembershipService.ChangePassword(this.User.Identity.Name, model.OldPassword, model.NewPassword))
                {
                    return this.RedirectToAction("ChangePasswordSuccess");
                }
                else
                {
                    this.ModelState.AddModelError(
                        "", "The current password is incorrect or the new password is invalid.");
                }
            }

            // If we got this far, something failed, redisplay form
            this.ViewBag.PasswordLength = this.MembershipService.MinPasswordLength;
            return View(model);
        }

        // **************************************
        // URL: /Account/ChangePasswordSuccess
        // **************************************

        public ActionResult ChangePasswordSuccess()
        {
            return this.View();
        }

        public ActionResult LogOff()
        {
            this.FormsService.SignOut();

            return this.RedirectToAction("Index", "Home");
        }

        // **************************************
        // URL: /Account/LogOn
        // **************************************

        public ActionResult LogOn()
        {
            return this.View();
        }

        [HttpPost]
        public ActionResult LogOn(LogOnModel model, string returnUrl)
        {
            if (this.ModelState.IsValid)
            {
                if (this.MembershipService.ValidateUser(model.UserName, model.Password))
                {
                    this.FormsService.SignIn(model.UserName, model.RememberMe);
                    if (this.Url.IsLocalUrl(returnUrl))
                    {
                        return this.Redirect(returnUrl);
                    }
                    else
                    {
                        return this.RedirectToAction("Index", "Home");
                    }
                }
                else
                {
                    this.ModelState.AddModelError("", "The user name or password provided is incorrect.");
                }
            }

            // If we got this far, something failed, redisplay form
            return View(model);
        }

        // **************************************
        // URL: /Account/LogOff
        // **************************************

        // **************************************
        // URL: /Account/Register
        // **************************************

        public ActionResult Register()
        {
            this.ViewBag.PasswordLength = this.MembershipService.MinPasswordLength;
            return this.View();
        }

        [HttpPost]
        public ActionResult Register(RegisterModel model)
        {
            if (this.ModelState.IsValid)
            {
                // Attempt to register the user
                var createStatus = this.MembershipService.CreateUser(model.UserName, model.Password, model.Email);

                if (createStatus == MembershipCreateStatus.Success)
                {
                    this.FormsService.SignIn(model.UserName, false /* createPersistentCookie */);
                    return this.RedirectToAction("Index", "Home");
                }
                else
                {
                    this.ModelState.AddModelError("", AccountValidation.ErrorCodeToString(createStatus));
                }
            }

            // If we got this far, something failed, redisplay form
            this.ViewBag.PasswordLength = this.MembershipService.MinPasswordLength;
            return View(model);
        }

        #endregion

        #region Methods

        protected override void Initialize(RequestContext requestContext)
        {
            if (this.FormsService == null)
            {
                this.FormsService = new FormsAuthenticationService();
            }
            if (this.MembershipService == null)
            {
                this.MembershipService = new AccountMembershipService();
            }

            base.Initialize(requestContext);
        }

        #endregion

        // **************************************
        // URL: /Account/ChangePassword
        // **************************************
    }
}

File: SAML 2 IdP-Initiated Web Example for ASP.NET MVC/[VB]-SAML 2 IdP-Initiated Web Example for ASP.NET MVC/VB/Saml2IdpInitiatedMvc.IdP/Controllers/AccountController.vb
Imports System.Diagnostics.CodeAnalysis
Imports System.Security.Principal
Imports System.Web.Mvc
Imports System.Web.Routing
Imports System.Web.Security
Imports Saml2IdpInitiatedMvc.Models

Namespace Saml2IdpInitiatedMvc.Controllers

	<HandleError> _
	Public Class AccountController
		Inherits Controller

		Public Property FormsService() As IFormsAuthenticationService
		Public Property MembershipService() As IMembershipService

		Protected Overrides Sub Initialize(ByVal requestContext As RequestContext)
			If FormsService Is Nothing Then
				FormsService = New FormsAuthenticationService()
			End If
			If MembershipService Is Nothing Then
				MembershipService = New AccountMembershipService()
			End If

			MyBase.Initialize(requestContext)
		End Sub

		' **************************************
		' URL: /Account/LogOn
		' **************************************

		Public Function LogOn() As ActionResult
			Return View()
		End Function

		<HttpPost> _
		Public Function LogOn(ByVal model As LogOnModel, ByVal returnUrl As String) As ActionResult
			If ModelState.IsValid Then
				If MembershipService.ValidateUser(model.UserName, model.Password) Then
					FormsService.SignIn(model.UserName, model.RememberMe)
					If Not String.IsNullOrEmpty(returnUrl) Then
						Return Redirect(returnUrl)
					Else
						Return RedirectToAction("Index", "Home")
					End If
				Else
					ModelState.AddModelError("", "The user name or password provided is incorrect.")
				End If
			End If

			' If we got this far, something failed, redisplay form
			Return View(model)
		End Function

		' **************************************
		' URL: /Account/LogOff
		' **************************************

		Public Function LogOff() As ActionResult
			FormsService.SignOut()

			Return RedirectToAction("Index", "Home")
		End Function
	End Class
End Namespace
